framework/include/utils/InputParametersChecksUtils.h

```diff
@@ -353,9 +358,43 @@ InputParametersChecksUtils<C>::checkVectorParamsNoOverlap(
 
     for (const auto & value : forwardGetParam<std::vector<T>>(param))
       if (!unique_params.insert(value).second)
+      {
+        auto copy_params = param_vec;
+        copy_params.erase(std::find(copy_params.begin(), copy_params.end(), param));
         forwardMooseError("Item '" + value + "' specified in vector parameter '" + param +
                           "' is also present in one or more of the parameters '" +
                           Moose::stringify(param_vec) + "'. This is disallowed.");
+      }
+  }
+}
+
+template <typename C>
+template <typename T>
+void
+InputParametersChecksUtils<C>::checkTwoDVectorParamsNoRespectiveOverlap(
+    const std::vector<std::string> & param_vec) const
+{
+  // Outer loop, each parameter is a vector of vector
```

Comment:
Wouldn't each entry in a range-based loop of a vector of string just be a string? What trickery is going on here?